kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -74,6 +74,9 @@ enum KaniAttributeKind {
     /// We use this attribute to properly instantiate `kani::any_modifies` in
     /// cases when recursion is present given our contracts instrumentation.
     Recursion,
+    /// Used to mark functions where generating automatic pointer checks should be disabled. This is
+    /// used later to automatically attach pragma statements to locations.
+    SkipPtrChecks,
```

Comment:
We might also want to disable other checks in this way in the future. Could you make this associated with a list of strings representing the checks we want to disable?
 
Also, I think the name `DisableChecks` will be more aligned with the name of disable-checks-pragmas in CBMC.

Reply:
I like the idea, but  I would avoid requiring the strings to match the ones from CBMC. In that case, make sure you document which strings are accepted

Reply:
We may want to use this later to disable Kani instrumentations

Reply:
Addressed that in the latest commit